gpii/node_modules/transformer/src/js/Transformer.js

```diff
@@ -47,10 +48,15 @@ var fluid = fluid || require("infusion"),
     gpii.transformer.transformOneSettingsHandler = function (settingsHandler, oneUserSolution, solutionId) {
         // TODO GPII-1223: throw app specific settings in the correct locations if there are multiple handlers
         var settings = fluid.copy(oneUserSolution.settings["http://registry.gpii.net/applications/" + solutionId]);
+        // extract any common terms that were defined in the application block
+        var scopedCommon = fluid.remove_if(fluid.copy(settings), function (val, key) {
+            return !key.startsWith("http://registry.gpii.net/common/");
+        });
 
         if (settingsHandler.capabilitiesTransformations) {
             var inferred = fluid.model.transformWithRules(oneUserSolution.settings, settingsHandler.capabilitiesTransformations);
-            settings = fluid.extend(true, inferred, settings);
+            var inferredScoped = (scopedCommon === undefined || $.isEmptyObject(scopedCommon)) ? {} : fluid.model.transformWithRules(scopedCommon, settingsHandler.capabilitiesTransformations);
```

Comment:
I don't think this guarding is necessary - the value can't be undefined because the settings can't be undefined, and if we transform an empty object it is harmless

Reply:
unfortunately not exactly (I'll file a JIRA about this) ... so this is the issue I was complaining about where e.g. UIO+'s capabilities transformation will result in the contrastTheme being set to default, if given an empty model... Since we prioritize scoped terms higher than top level terms, this will result in the contrastTheme being set to "default" even if it's explicitly set to something in a top-level common term

Reply:
Did you file the JIRA in the end?

Reply:
yes, it's GPII-2626